integrations/google-analytics-4/lib/index.js

```diff
@@ -0,0 +1,303 @@
+    /**
+     * Measurement IDs.
+     * The same configuration information is shared across all measurement IDs.
+     * https://developers.google.com/analytics/devguides/collection/ga4#add_an_additional_google_analytics_property_to_an_existing_tag
+     */
+    for (var i = 0; i < measurementIds.length; i++) {
+      window.gtag('config', measurementIds[i], config)
+
+    }
+
+    /**
+     * Set persistent values shared across all gtag.js usage.
+     * https://developers.google.com/gtagjs/reference/api#set
+     */
+    for (var i = 0; i < sets.length; i++) {
+      window.gtag.apply(null, sets[i]);
+    }
+
+    self.ready();
+  });
+};
+
+/**
+ * Loaded?
+ *
+ * @api private
+ * @return {boolean}
+ */
+GA4.prototype.loaded = function() {
+  return !!(
+    window.ga4DataLayer && Array.prototype.push !== window.ga4DataLayer.push
+  );
+};
+
+/**
+ * Identify.
+ *
+ * @api public
+ * @param {Facade.Identify} event
+ */
+GA4.prototype.identify = function(identify) {
+  var opts = this.options;
+  var userPropertyMappings = opts.userProperties;
+
+  var userProperties = {};
+
+  // Map all customer-defined user property mappings.
+  for (var eventField in userPropertyMappings) {
+    if (!userPropertyMappings.hasOwnProperty(eventField)) {
+      continue;
+    }
+
+    var userProp = userPropertyMappings[eventField];
+    var value = identify.proxy(eventField);
+
+    userProperties[userProp] = value;
+  }
+
+  /**
+   * Map the user_id property if the Google Reporting Identity is set one of:
+   *   - By User ID, Google signals, then device (userIdSignalsAndDevice)
+   *   - By User ID and Devicea (userIdAndDevice)
+   * 
+   * Google's Reporting Identity: https://support.google.com/analytics/answer/9213390?hl=en
+   * 
+   * Note that the user ID can be appended as part of the user_properties
+   * object instead of being configured by an explicit command.
+   * https://developers.google.com/analytics/devguides/collection/ga4/cookies-user-id#set_user_id
+   */
+  var userId = identify.userId();
+  var validReportingIdentity = opts.googleReportingIdentity === 'userIdSignalsAndDevice' || opts.googleReportingIdentity === 'userIdAndDevice'
+  if (userId && validReportingIdentity) {
+    userProperties.user_id = userId;
+  }
+
+  if (Object.keys(userProperties).length) {
+    window.gtag('set', 'user_properties', userProperties);
+  }
+};
+
+/**
+ * Group
+ *
+ * @api public
+ * @param {Facade.Group} group
+ */
+GA4.prototype.group = function(group) {
+  window.gtag('event', 'join_group', {
+    group_id: group.groupId()
+  });
+};
+
+/**
+ * Page
+ *
+ * @api public
+ * @param {Facade.Page} page
+ */
+GA4.prototype.page = function(page) {
+  // If the Send Google's Automatic Page View Measurement setting is set to true then
+  // don't handle page calls to avoid duplicate page_view events.
+  if (this.options.sendAutomaticPageViewEvent) {
+    return;
+  }
+
+  // Track categorized pages setting needed here?
```

Comment:
Have we been able to look into this? :)

Reply:
I ran it by Jen but no response yet. I think we can leave it out for now and add it if customers request it. What do you think?

Reply:
Jen recommended we omit it until this functionality is requested.